### PR TITLE
feat(licenseinfo): store selection and display usage

### PR DESCRIPTION
**Pull request by lepokle**
The selected license files and licenses are now automatically stored on license info download.
Next time, this stored selection is used to preselect the licenses.

Closes sw360/sw360portal#580
Closes sw360/sw360portal#581

### CODE REVIEW

**Comment by lepokle** [comment]
**This PR cannot be tested without merging #599**

**Comment by ChristophNiehoff** [comment]
rebased to current master

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/LinkedReleasesAndProjectsAwarePortlet.java [hunk 1]
```diff
@@ -91,17 +95,18 @@ protected void putDirectlyLinkedReleasesInRequest(PortletRequest request, Projec
         request.setAttribute(RELEASE_LIST, linkedReleases);
     }
 
-    protected void putLinkedProjectsInRequest(PortletRequest request, Project project, User user) {
-        putLinkedProjectsInRequest(request, project, Function.identity(), user);
+    protected List<ProjectLink> createLinkedProjects(Project project, User user) {
```
Comment (ChristophNiehoff) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/LinkedReleasesAndProjectsAwarePortlet.java]: Ok, one can do this. But is there a deeper reason for abandoning `putLinkedProjectsInRequest`?  Is it only to have access to `mappedProjectLinks` in line 766 of the `ProjectPortlet`?  Wouldn't it be more readable to have a `createLinkedProjects` function as well as a `putLinkedProjectsInRequest` function? Then one wouldn't have to replace every occurrence of `putLinkedProjectsInRequest` by two lines all over the code.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/attachmentSelectTable.jspf [hunk 2]
```diff
@@ -14,12 +14,23 @@
     <portlet:param name="<%=PortalConstants.ACTION%>" value='<%=PortalConstants.GET_LICENCES_FROM_ATTACHMENT%>'/>
 </portlet:resourceURL>
 
+<portlet:resourceURL var="downloadLicenseInfoSatetURL">
```
Comment (ChristophNiehoff) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/attachmentSelectTable.jspf]: typo?  `downloadLicenseInfoStateURL`?

**Comment by ChristophNiehoff** [comment]
Changed the remarked styling issues.

During that I noticed a small bug that I overlooked in my code-review:
In line 184 of `ProjectProtlet.java` the line 
```
if (project.getPermissions() == null || project.getPermissions().get(RequestedAction.WRITE))
```
was used to determine if the use has write permissions for the project. However, `ProjectDatabaseHandler.getProjectById` does not set the permissions in the `project` object, which is why the above condition is always true. Changed this line to the explicit form
```
if (PermissionUtils.makePermission(project, user).isActionAllowed(RequestedAction.WRITE))
```

**Comment by mcjaeger** [comment]
testing has revealed that usage at attachment listing is not updated, but I think this is OK for now

**Comment by ChristophNiehoff** [comment]
Implemented the change of message.

**Pull request merged**